### ISSUE TITLE
Can't move/manage forum order

### ISSUE THREAD

**Issue by wrxxy** [e107inc/e107#3545]
Everytime i try to move/manage forum/subforums orders they stop working with v2 and link back to forum, it used to work with v1 but no longer..... I am not sure if this has to do with my v1 to v2 update but the problem has been there since update.. not sure how to fix as I am no guru but just wondering if anyone can help.... there may be other issue's opened due to this

![image](https://user-images.githubusercontent.com/13065217/48174154-e39bcb00-e35a-11e8-87cc-c1ad9ef36d47.png)

**Comment by wrxxy** [comment]
Not sure how to explain but when I move a parent forum should'nt the subforums under those move also?
as this doesn't happen. I get no errors its just I am not able to access those forum, as link, links back to main forum. That's probably my best explanation...

**Comment by wrxxy** [comment]
You actually need to see it while trying to manage forum order...

**Comment by Moc** [comment]
1. Subforums do not move together with parent. 
Confirmed. Seems like a bug, or missing functionality. @Deltik Any thoughts on this? Not developed yet, or bug? 

2. Links not working. 
Cannot reproduce yet. You mean the link on the frontend of your website? What is the URL it links to?

**Comment by wrxxy** [comment]
Yes frontend links back to main forum from memory....

**Comment by Moc** [comment]
So it would be interesting to know what the actual link is that is rendered. Is your website public so I can see what's going on?

**Comment by wrxxy** [comment]
yes its up www.outlawsracing.com i have contacted you on gitter....

**Comment by Moc** [comment]
I'm limiting this issue to just the first problem described above. 
The issue of the links not working is discussed here: https://github.com/e107inc/e107/issues/3521

**Comment by Deltik** [comment]
@Moc: It looks like the forum parent-child relationship structure changed from e107 v0.8 to e107 v2.1, and now, the structure is stuck in a limbo where items can have a "Parent" parent and a "Sub-forum" parent.  These aren't validated and a forum can be configured to have contradictory parents by moving the forums' inheritances around.

Walking back in time, I noticed that the tree model display format has never coped with this multi-parent concept correctly.

Apparently, the forum plugin has not ever abstracted forum items properly, but in e107 v0.7, it was hacked to work by having "Parent" parents sit at the top, "Forum" parents sit at the second level, and "Sub-forums" fill the third level.

**Comment by Moc** [comment]
For me now the subforums appear to be moving with the parent. 

E.g. 

> Parent 1
>> Child 1.1 
>>> Subforum 1.1.1

If i change the order of Parent 1, with another Parent (e.g. Parent 2), both Child 1.1. and Subforum 1.1.1 seem to move correctly along. 

Were there any changes on this recently?

**Comment by Deltik** [comment]
@Moc: In your example structure, "Child 1.1" should actually be the child of "Subforum 1.1.1".

Now, it is true that you can change the order by dragging the ![re-order](https://user-images.githubusercontent.com/1364268/52457445-616c9f80-2b1f-11e9-868c-fff550dd6e5b.png) bar, but even though it's changing in the backend, the children are not being reordered with it.

To make things more confusing, if "Child 1.1" does not have matching "Parents" and "Sub-forum" fields or even if the "Parents" field is omitted, it will be represented confusingly in the tree list.

**Comment by Moc** [comment]
@Deltik I just came across this when testing a v1 forum upgrade to v2. 

On v1, I created Parent 1, Child 1.1 and Subforum 1.1.1. In v1, there is only one way to add a subforum, and that is to click on the forum (1.1), so I am sure that this is the right hierarchy.

This is how it looks after the v2 update:

All forums:
![image](https://user-images.githubusercontent.com/577978/59041293-98a64f80-8878-11e9-8c3d-733b870c7836.png)

Subforum 1.1.1:
![image](https://user-images.githubusercontent.com/577978/59041343-afe53d00-8878-11e9-8199-d30f518ac9d7.png)

I then used the blue arrows to drag and change the order of the subforums (changing around 1.1.1 and 1.1.2). This works fine. 

Now, when I move child 1.1 and 1.2 around. The issue starts occurring:

![image](https://user-images.githubusercontent.com/577978/59041537-15d1c480-8879-11e9-9338-d8673baa3a86.png)


The subforums 1.1.1 and 1.1.2 are not moving together with child 1.1. 

The frontend still works fine: 
![image](https://user-images.githubusercontent.com/577978/59041620-41ed4580-8879-11e9-92b1-0e1efda1c419.png)

Hope this helps to illustrate the problem

**Comment by CaMer0n** [comment]
@Moc This will likely be solved with changes to: `public function afterSort($result, $selected)` in **forum_admin.php** line 317. Ie. check if the moved item has children and update their values in the db.
